cds_ils/ldap/api.py

```diff
@@ -146,196 +135,254 @@ class LdapUserImporter:
+    command = "ils patrons index"
+    runner.invoke(cli, command, catch_exceptions=True)
+
+
+def import_ldap_users():
     """Import ldap users in db."""
+    start_time = time.time()
+
+    ldap_client = LdapClient()
+    ldap_users = ldap_client.get_primary_accounts()
 
-    def index_ldap_users():
-        """Index ldap users in ES."""
-        from invenio_base.app import create_cli
+    print("Users in LDAP: {}".format(len(ldap_users)))
 
-        cli = create_cli()
-        runner = current_app.test_cli_runner()
-        command = "ils patrons index"
-        runner.invoke(cli, command, catch_exceptions=True)
-        _log_info("command_executed", dict(command=command))
+    importer = LdapUserImporter()
+    for ldap_user in ldap_users:
+        email = ldap_user_get(ldap_user, "mail").lower()
+        if User.query.filter_by(email=email).count() > 0:
+            print(
+                "User with email {} already imported, skipping.".format(email)
+            )
+            continue
 
-    importer = LdapUserImporter(ldap_users)
-    importer.import_users()
-    index_ldap_users()
-    _log_info("users_indexed")
+        employee_id = ldap_user_get(ldap_user, "employeeID")
+        print("Importing user with person id {}".format(employee_id))
+        importer.import_user(ldap_user)
 
+    db.session.commit()
 
-def check_user_for_update(system_user, ldap_user):
-    """Check if there is an ldap update for a user and commit changes."""
-    ldap_user_department = ldap_user["department"][0].decode("utf8")
-    if not system_user.extra_data.get("department") == ldap_user_department:
-        previous_department = system_user.extra_data.get("department")
+    _reindex_patrons()
 
-        system_user.extra_data["department"] = ldap_user_department
-        db.session.commit()
+    print("Users imported: {}".format(len(ldap_users)))
+    print("--- Finished in %s seconds ---" % (time.time() - start_time))
 
-        _log_info(
-            "department_updated",
-            dict(
-                user_id=system_user.user.id,
-                previous_department=previous_department,
-                new_department=ldap_user_department,
-            ),
-        )
+
+def _update_invenio_user(invenio_remote_account_id, ldap_user):
+    """Check if the LDAP user has more updated info and update the Invenio."""
+    ra = RemoteAccount.query.filter_by(id=invenio_remote_account_id).one()
+    ra.extra_data["department"] = ldap_user_get(ldap_user, "department")
+    db.session.commit()
 
 
-def delete_user(user_id):
-    """Delete a system user."""
-    with current_app.app_context():
-        try:
-            anonymize_patron_data(user_id)
-        except AssertionError:
-            send_active_loans_mail(user_id)
+def _delete_invenio_user(user_id):
+    """Delete an Invenio user."""
+    try:
+        anonymize_patron_data(user_id)
+        return True
+    except AssertionError:
+        send_active_loans_mail(user_id)
```

Comment:
@jrcastro2 you will have to refactor this line, just be aware of the changes, start from the PR and not from master